web/react/components/textbox.jsx

```diff
@@ -27,11 +40,48 @@ module.exports = React.createClass({
     },
     componentWillUnmount: function() {
         PostStore.removeAddMentionListener(this._onChange);
+        ErrorStore.removeChangeListener(this._onError);
     },
     _onChange: function(id, username) {
         if (id !== this.props.id) return;
         this.addMention(username);
     },
+    _onError: function() {
+        var errorState = getStateFromStores();
+
+        if (this.state.timerInterrupt != null) {
+            window.clearInterval(this.state.timerInterrupt);
+            this.setState({ timerInterrupt: null });
+        }
+
+        if (errorState.message === "There appears to be a problem with your internet connection") {
+            this.setState({ connection: " bad-connection" });
```

Comment:
Looks good other than the space in " bad-connection". I understand that you're adding it so that the className of the textarea below is properly formatted, but I'd just add the extra space directly to the HTML. Having a space in what is essentially an enum feels odd to me, and having potentially an extra space in the HTML isn't much of a concern.